charts/vela-core/templates/defwithtemplate/config-dex-connector.yaml

```diff
@@ -4,7 +4,7 @@ apiVersion: core.oam.dev/v1beta1
 kind: ComponentDefinition
 metadata:
   annotations:
-    custom.definition.oam.dev/name.config.oam.dev: Dex Connectors
+    custom.definition.oam.dev/name.config.oam.dev: Dex Connector
```

Comment:
```suggestion
    custom.definition.oam.dev/config-name: Dex Connector
```

Reply:
the `oam.dev` is duplicated twice, please avoid this case

Reply:
I want to store the friendly name `Dex Connector` as a label `custom.definition.oam.dev/catalog.config.oam.dev: Dex Connector` like blow.
```
  labels:
    custom.definition.oam.dev/catalog.config.oam.dev: velacore-config
    custom.definition.oam.dev/multi-cluster.config.oam.dev: "false"
    custom.definition.oam.dev/type.config.oam.dev: dex-connector
```
But k8s constraints don't allow it. Then I set it an annotation. If followed the suggestions, how to express the config is `config.oam.dev` instead of a common label?

Reply:
`custom.definition.oam.dev` is automatically added. It would be friendly to review raw .cue file https://github.com/oam-dev/kubevela/pull/3430/files#diff-afd5da4869a59d34f9f87fc352e178548ba7c03490b169623ce3d2c7b1e0ace8.